example/spec/ingestGranule/IngestGranuleSuccessSpec.js

```diff
@@ -276,4 +281,46 @@ describe('The S3 Ingest Granules workflow', () => {
       expect(record.status).toEqual('completed');
     });
   });
+
+  describe('encounters duplicate filenames', () => {
+    let fileUpdated;
+    let secondWorkflowExecution;
+
+    beforeAll(async () => {
+      // update one of the input files so we can assert that the file size changed
+      const content = randomString();
+      const file = inputPayload.granules[0].files[0];
+      fileUpdated = file.name;
+      const updateParams = {
+        Bucket: config.bucket, Key: path.join(file.path, file.name), Body: content
+      };
+
+      await s3().putObject(updateParams).promise();
+      inputPayload.granules[0].files[0].fileSize = content.length;
+
+      secondWorkflowExecution = await buildAndExecuteWorkflow(
+        config.stackName, config.bucket, workflowName, collection, provider, inputPayload
+      );
+    });
+
+    it('does not raise a workflow error', () => {
+      expect(secondWorkflowExecution.status).toEqual('SUCCEEDED');
+    });
+
+    it('overwrites the existing file with the new data', async () => {
+      const lambdaOutput = await lambdaStep.getStepOutput(secondWorkflowExecution.executionArn, 'MoveGranules');
+      const outputFiles = lambdaOutput.payload.granules[0].files;
+      const currentFiles = await getFilesMetadata(outputFiles);
+
+      expect(currentFiles.length).toBe(existingFiles.length);
```

Comment:
You're totally right. I was having some difficulty integrating my new test into this spec, because since my test changes one of the file sizes to assert the change was definitively made, it fails the assertion for `has expected payload` in `PostToCmr` if it runs before that. 

Using the lambda output from the `MoveGranules` task seemed like the easiest way to get existing files from the first run of the workflow before they are deleted in the cleanup. But I failed to recognize that the fact the files are cleaned up means that my test isn't really asserting overwrite at all. I'll look into how I can refactor